lib/utils/combineJSON.js

```diff
@@ -35,12 +36,13 @@ function combineJSON(arr, deep, collision) {
   }
 
   for (i = 0; i < files.length; i++) {
-    var file_content = fs.readFileSync(files[i], 'utf8');
+    var resolvedPath = resolveCwd(path.isAbsolute(files[i]) ? files[i] : '.' + path.sep + files[i]);
+    var file_content;
 
     try {
-      file_content = JSON.parse(file_content);
+      file_content = require(resolvedPath);
     } catch (e) {
-      throw new SyntaxError(files[i] + ' - failed to parse JSON: ' + e.message);
+      throw new SyntaxError(files[i] + ' - failed to load or parse JSON or JS Object: ' + e.message);
```

Comment:
I noticed one other thing here.. by wrapping the `e` in a SyntaxError it loses some of the details about the actual error. The message is there, but accurate file and line number go missing. Any objections to just throwing the actual `e` here and maybe just maybe prepending the 'failed to load or parse JSON or JS Object' boilerplate to the actual e's message? The difference:

Throw the `e`:

```
TypeError: Cannot read property 'dimension' of undefined
    at tokens (/Users/e828613/sc/manhattan/mds-tokens/src/tokens/decisions/btn/btn-primary.js:4:40)
    at Object.<anonymous> (/Users/e828613/sc/manhattan/mds-tokens/src/tokens/decisions/btn/btn-primary.js:20:4)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at combineJSON (/Users/e828613/sc/manhattan/mds-tokens/node_modules/style-dictionary/lib/utils/combineJSON.js:43:22)
```

Existing (with a different but similar underlying error) doesn't have the detail in the stack...

```
SyntaxError: src/tokens/decisions/btn/btn-primary.js - failed to load or parse JSON or JS Object: Cannot read property 'none' of undefined
    at combineJSON (/Users/e828613/sc/manhattan/mds-tokens/node_modules/style-dictionary/lib/utils/combineJSON.js:45:13)
    at Object.extend (/Users/e828613/sc/manhattan/mds-tokens/node_modules/style-dictionary/lib/extend.js:113:17)
    at Object.<anonymous> (/Users/e828613/sc/manhattan/mds-tokens/src/dictionary-config/style-dictionary.js:8:57)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:188:16)
```

Reply:
Very good point, yes we should throw the e, but make sure the message for JSON files shows the filename. We redid the errors a few weeks ago to be more descriptive so it is easier to debug the issue, and adding the filename helps a lot (it previously just had a generic JSON error with no filename). So this fits perfectly in that theme of useful error messages.